### DISCUSSION TITLE
Airflow 2.9.1 / CentOS 7 / google-re2

### DISCUSSION THREAD

**Discussion by AlekseiSaff** [apache/airflow#39649] [General]
### Apache Airflow version

2.9.1

### What happened?

It seems that requirement of google-re2==1.1.20240501 broke installation of airflow 2.9.1 on CentOS7

two issues: 

- pybind11 not found - easy to fix via manual installation (might be needed also to be presented in https://raw.githubusercontent.com/apache/airflow/constraints-2.9.1/constraints-3.9.txt)

- the second problem is more complex. google-re2==1.1.20240501 requires presence of absl which is not in the system. I tried to compile the latest version of absl-cpp, but seems like it's incompatible with re2-devel from CentOS 7

 

### What you think should happen instead?

centos 7 still should be supported and it seems google-re2==1.1 still works with airflow 2.9.1

### How to reproduce

try to install google-re2==1.1.20240501 on CentOS7 

### Operating System

CentOS Linux release 7.9.2009 (Core)

### Deployment

Virtualenv installation

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by AlekseiSaff** [comment]
https://github.com/google/re2/issues/489 
https://github.com/google/re2/commit/f56cbe21b72dcddc91f84139b7434cc09b7fdf99

so it means 2.9.1 and onward won't be suported on CentOS 7.x

**Comment by potiuk** [comment]
@AlekseiSaff . To be perfectly honest I have hard time to understand your requesr. Why do you think that 

> centos 7 still should be supported and it seems google-re2==1.1 still works with airflow 2.9.1 

CentOS 7 is EOL end of Jun/ e. If you think that the whole community should be held-back so that you can keep Airflow running :) . That seems pretty bold statement. Probably better is - if you want to continue runing CentOS 7, you should pay to companies who offer extended support and make it compliant with Airflow? Would not that be more reasonable ask ? 

Or simply move to a supported CentOS version maybe if you do not want to pay ?
